packages/table/src/Table.ts

```diff
@@ -238,7 +238,7 @@ export class Table extends SizedMixin(SpectrumElement, {
                 this.tableHeadCheckboxCell = document.createElement(
                     'sp-table-checkbox-cell'
                 ) as TableCheckboxCell;
-                this.tableHead.insertAdjacentElement(
+                this.tableHead?.insertAdjacentElement(
```

Comment:
If this code runs, and it doesn't do the insert because it ran too early in whatever context, then it won't run again and the `tableHeadCheckboxCell` element will never be inserted into the `tableHead`. Does you use case (in the S2 docs) actually use a Table Head? If it works for you there, then I'm just being over worried, but the test you added (while great) doesn't leverage the alternative timing that you would see in React confirm this.

Reply:
Hi @Westbrook,

I understand your concern regarding the potential risk of the tableHeadCheckboxCell element not being inserted into the tableHead due to early execution.

To address your question: Yes, in the S2 docs use case that I'm working on, the table does indeed have a header. I tested this change locally, and it worked as expected. Interestingly, while running S2 docs in local dev mode, I noticed that the willUpdate() function always gets invoked twice. The first invocation happens before the table headers are populated. However, with the inclusion of optional chaining, by the second invocation, everything proceeds smoothly and as intended.

The new test case I added, though not perfect, is the closest approximation I could come up with to simulate this scenario. I believe it's a good representation of the behavior I observed locally.

I hope this offers some clarity. Always appreciate your diligent feedback!